engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -122,7 +122,7 @@ def verify_base_product_upgrade
 
           activated_bases = @system.products.where(product_type: 'base')
           activated_bases.each do |base_product|
-            return true if (base_product.identifier == upgrade_product.identifier)
+            return true if base_product.product_class == upgrade_product.product_class
```

Comment:
Maybe `base_product.identifier == upgrade_product.identifier || base_product.product_class == upgrade_product.product_class` is more safe.

Reply:
I considered that option but isn't  `product_identifer` matching a subset of `product_class` matching ?

Reply:
The product identifier can change across versions (as happened with Micro), but also the product class can change. So the suggested condition would be more safe. 
In the end what we do here is a hack because it's not based on the subscription. In SCC, the migration would check if the target product is included in the system's subscription.

Reply:
OK, so, similar to the issue with SUMA ARM64, a better check on subscription is recommended/encouraged 

I will do that instead of the double hack (product identifier + product class)

Thoughts ?

Reply:
In general, it would be useful if you have a way also for PAYG to find out the used subscription, to use it's settings to determine which products a system should have access to.

Reply:
yes, the fix/new check would be for PAYG as well